vcsh

```diff
@@ -146,11 +149,13 @@ help() {
 # editorconfig-checker-enable
 
 debug() {
-	if [ -n "$VCSH_DEBUG" ]; then echo "$SELF: debug: $@"; fi
+	if [ -n "$VCSH_DEBUG" ]; then echo "$SELF: debug: $*"; fi
```

Comment:
I will need to read up on `$@` vs `$*` again

Reply:
`$@` is an array, `$*` is a string. And generally you only want to use them in double quotes. In fact, I can't think of a single use case for an unquoted `$@`.

```
cat foo.sh             
#!/bin/bash

for i in "$*"; do
  echo "\"\$\*\": $i"
done
for i in $*; do
  echo "\$\*: $i"
done
for i in "$@"; do
  echo "\"\$@\": $i"
done
for i in $@; do
  echo "\$@: $i"
done
$ ./foo.sh 1 2 3 "4  5" 6
"$\*": 1 2 3 4  5 6
$\*: 1
$\*: 2
$\*: 3
$\*: 4
$\*: 5
$\*: 6
"$@": 1
"$@": 2
"$@": 3
"$@": 4  5
"$@": 6
$@: 1
$@: 2
$@: 3
$@: 4
$@: 5
$@: 6
```

Reply:
POSIX sh doesn't have arrays. Your examples here are Bash based.

Reply:
POSIX shell doesn't have arrays but it does draw a distinction between `$*` and `$@`. It's where `bash` got its array syntax from. My example would work the same in dash (`/bin/sh` on Ubuntu) or on a posix shell on a BSD or vendoer Unix system.

https://pubs.opengroup.org/onlinepubs/9699919799/utilities/V3_chap02.html#tag_18_05_02

Reply:
My bad, my comment is only half right. POSIX doesn't specify have arrays, it specifies array—singular. `$@` is the only available one.